src/python/zquantum/core/wip/circuit/conversions/cirq_conversions.py

```diff
@@ -0,0 +1,235 @@
+"""Utilities for converting gates and circuits to and from Cirq objects."""
+from functools import singledispatch
+from typing import Union, Type, Callable
+
+import cirq
+import numpy as np
+import sympy
+
+from ...circuit.gates import (
+    Gate,
+    X,
+    Y,
+    Z,
+    RX,
+    RY,
+    RZ,
+    PHASE,
+    T,
+    I,
+    H,
+    CZ,
+    CNOT,
+    CPHASE,
+    SWAP,
+    XX,
+    YY,
+    ZZ,
+    XY,
+)
+
+Param = Union[sympy.Expr, float]
+RotationGateFactory = Callable[[Param], cirq.EigenGate]
```

Comment:
Should this be `Callable[List[Param], cirq.EigenGate]`?

Reply:
The `[]` on first position in `Callable` type hint is actually a _list of the callable's argument types_, not a Python list, so this type annotation looks legit.